service/history/conflictResolver.go

```diff
@@ -106,6 +106,7 @@ func (r *conflictResolver) reset(replayEventID int64, startTime time.Time) (*mut
 	resetMutableStateBuilder.executionInfo.StartTimestamp = startTime
 	// the last updated time is not important here, since this should be updated with event time afterwards
 	resetMutableStateBuilder.executionInfo.LastUpdatedTimestamp = startTime
+	resetMutableStateBuilder.updateReplicationStateLastEventID(sourceCluster, replayEventID)
```

Comment:
I think just setting the lastWriteEventID won't be enough.  We also need to set the lastWriteVersion correctly too.

Reply:
those are updated in #806